torch/_inductor/codegen/cpp.py

```diff
@@ -268,6 +269,8 @@ def constant(val, dtype):
             quote = f"std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::infinity()"
         elif val == float("-inf"):
             quote = f"-std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::infinity()"
+        elif math.isnan(val):
+            return f"std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::quiet_NaN()"
```

Comment:
```suggestion
            quote = f"std::numeric_limits<{DTYPE_TO_CPP[dtype]}>::quiet_NaN()"
```
Please add the case to cover it too. Code sequence like below can trigger this vec path:
```python
    a = torch.tensor(float("nan"))
    b = torch.empty([100])
    b.fill_(a)
```